src/Farmer/Builders/Builders.Storage.fs

```diff
@@ -52,7 +52,10 @@ type StorageAccountConfig =
     /// Gets the ARM expression path to the key of this storage account.
     member this.Key = StorageAccount.getConnectionString this.Name
     /// Gets the Primary endpoint for static website (if enabled)
-    member this.WebsitePrimaryEndpoint = sprintf "https://%s.z6.web.core.windows.net" this.Name.ResourceName.Value
+    member this.WebsitePrimaryEndpoint =
+        ArmExpression
+            .create(sprintf "reference(%s, '2019-04-01').primaryEndpoints.web" this.Name.ResourceName.Value)
+            .Eval()
```

Comment:
I would even consider returning this as an ARM Expression and let the consumer decide when they want to Eval it.